src/rust/src/contract.rs

```diff
@@ -168,9 +168,9 @@ fn non_metered_xdr_from_cxx_buf<T: ReadXdr>(buf: &CxxBuf) -> Result<T, HostError
 
 fn non_metered_xdr_to_rust_buf<T: WriteXdr>(t: &T) -> Result<RustBuf, HostError> {
     let mut vec: Vec<u8> = Vec::new();
-    t.write_xdr(&mut xdr::DepthLimitedWrite::new(
+    t.write_xdr(&mut xdr::Limited::new(
         Cursor::new(&mut vec),
-        MARSHALLING_STACK_LIMIT,
+        Limits {depth: MARSHALLING_STACK_LIMIT, len: 0x1000000},
```

Comment:
This is hard to interpret and also probably too much. I would probably set this to 5 MB and write that clearly as `5 * 1024 * 1024`

Reply:
Is 5MB enough as an upper bound?

Reply:
I'm thinking specifically about WASM size

Reply:
The LE size limit is 64K, so we're not even close to that limit.
More importantly, this is non-metered function. It is only used in two contexts now: for diagnostics (non-observable, can be updated without protocol change if ever necessary), and for TTL entry encoding (small bounded size, though even that seems a bit iffy). Given that it's non-metered, I don't think we should use it to convert any non-constant-size observable data.

Reply:
Updated